### DISCUSSION TITLE
yamlpath syntax for appending to a list

### DISCUSSION THREAD

**Discussion by floli** [wwkimball/yamlpath#161] [Q&A]
Hello,

I use yamlpath from python code, mainly to manipulate, like `processor.set_value(path, val)`. I wasn't able to find out how to append to a list, without knowing it's length beforehand. What is the correct yamlpath syntax for that?

Thanks!

**Comment by wwkimball** [comment]
Many people think of adding an element to an array as a simple scalar operation; with YAML, it is vastly more complex due to YAML's native merge capabilities.  Please refer to the following discussions for a deep-dive into why the Merger class (implemented in the `yaml-merge` command) is the recommended tool for adding (merging) data into YAML Arrays:

  * https://github.com/wwkimball/yamlpath/issues/118
  * https://github.com/wwkimball/yamlpath/issues/141

This documentation covers the Merger class and its reference implementation, the `yaml-merge` command-line tool:

  * https://github.com/wwkimball/yamlpath/wiki/Python-Library#merging-documents
  * https://github.com/wwkimball/yamlpath/wiki/merger.py
  * https://github.com/wwkimball/yamlpath/wiki/yaml-merge
  * https://github.com/wwkimball/yamlpath/wiki/yaml-merge-Examples

That said, I'm always interested in making YAML Path better.  The closest we've come to an idea for how to allow scalar, atomic additions to Arrays was a proposal to define a new YAML Path segment but that idea quickly fell apart.  If you have a good answer for this question in #141, I'd be happy to explore it with you:

> In another light, adding a new YAML Path segment like [+] or [*] must be considered in the greater context of what such a segment would mean to other use-cases for YAML Paths. What should the yaml-get command do when it receives a YAML Path like "some.path.to[+]"?
